contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -680,10 +680,11 @@ class Resource extends mix(APIResource, IndexedDBResource) {
       /* eslint-enable */
     }
     return this.table.get(id).then(obj => {
+      const request = this.requestModel(id);
       if (obj) {
         return obj;
       }
-      return this.requestModel(id);
+      return request;
```

Comment:
Could you explain the difference here?

Reply:
We are now making the request, regardless of whether the data was found in IndexedDB or not, so we serve the stale data in the mean time, but invalidate the stale cache with the request.

Reply:
I know some of the copy and move commands are fairly rapid-fire, so this could produce a lot of requests. I'm not sure of the impact but combined with the other recent change I made, the net result could not be so great.  This doesn't necessarily help that, but what if in the case when we do have a copy locally, we instead put the `requestModel` call in the next tick?

Reply:
I think the overall effect of the indirection in `requestModel` does mean that it gets shunted out of the main thread and picked up later. I could also add the same sort of request sharing that we are using for `requestCollection` where we maintain the fetch promise for 5 seconds and just pass it back if the same request is made again.